src/ssa_tracker.rs

```diff
@@ -46,9 +48,11 @@ impl SSATracker {
                 let skipped_idx = self.current[key];
                 if skipped_idx != *value {
                     let new_ssa = self.allocate_new_ssa_index_by_id(*key);
+                    // TODO This does not work for memory
                     Some(format!(
-                        "(define-const {} (_ BitVec 256) (ite {} {} {}))\n",
+                        "(define-const {} {} (ite {} {} {}))\n",
                         self.id_to_smt_variable(*key, new_ssa).name,
+                        self.type_of_id(*key),
```

Comment:
Isn't this returning only BV8 for now anyway?

Reply:
It's BV256 for all variables except the memory variable for which it returns an array type

Reply:
where does that happen?

Reply:
https://github.com/leonardoalt/yools/pull/66/files#diff-ab01308940caaee595a2da5f1569b16a0fd60fa02c7cb18e5d8d71b4e0918da8R27